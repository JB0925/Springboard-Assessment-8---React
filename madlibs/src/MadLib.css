.MadLib {
    font-size: 2rem;
}