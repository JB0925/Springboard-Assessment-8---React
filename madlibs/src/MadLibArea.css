.MadLibArea {
    width: 20%;
    display: flex;
    align-items: center;
    justify-content: center;
    flex-direction: column;
}

p {
    text-align: center;
}

.MadLibArea-MadLib {
    width: 5%;
    padding: 5px;
}