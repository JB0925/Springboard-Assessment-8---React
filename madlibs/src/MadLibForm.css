.MadLibForm {
    display: flex;
    flex-direction: column;
    align-items: flex-start;
    justify-content: center;
    padding: 25px;
    border: .5px solid lightgray;
    width: 100%;
    box-shadow: rgba(17, 12, 46, 0.15) 0px 48px 100px 0px;
}

input, label, button{
    margin-top: 10px;
}

input {
    width: 80%;
    border-radius: 5px;
    padding: 5px;
}

button {
    background-color: dodgerblue;
    border-radius: 5px;
    color: white;
    padding: 5px;
    width: 40%;
}
