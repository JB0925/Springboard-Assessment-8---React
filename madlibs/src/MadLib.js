import React, { useState } from "react";
import "./MadLib.css";

/** A simple component used to display the Madlib
 *  generated from the form.
 */
const MadLib = ({ message }) => {
    return (
        <p className="MadLib">{message}</p>
    );
};

export default MadLib;