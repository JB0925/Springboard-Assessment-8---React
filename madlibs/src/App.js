import MadLibArea from './MadLibArea';
import './App.css';

function App() {
  return (
    <div className="App">
      <h1>Madlibs!</h1>
      <MadLibArea />
    </div>
  );
}

export default App;
