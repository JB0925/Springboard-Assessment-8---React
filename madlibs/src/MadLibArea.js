import React, { useState } from "react";
import { useMadLibUpdate, useChangeSubmitted } from "./Hooks/hooks";
import MadLib from "./MadLib";
import MadLibForm from "./MadLibForm";
import "./MadLibArea.css";

/** The area used to display either the form or the MadLib,
 *  depending on the value of "isSubmitted".
 */
const MadLibArea = () => {
    const [madLib, addMadLib] = useMadLibUpdate();
    const [isSubmitted, updateSubmitted] = useChangeSubmitted();

    const { noun, nounTwo, adjective, color } = madLib;
    const message = `There was a ${color} ${noun} who loved a ${adjective} ${nounTwo}.`;

    return ( 
        !isSubmitted ?
        <div className="MadLibArea">
            <MadLibForm addMadLib={addMadLib} updateSubmitted={updateSubmitted} />
        </div>
        :
        <div className="MadLibArea" style={{width: "100%"}}>
            <MadLib message={message} />
            <button className="MadLibArea-MadLib" onClick={updateSubmitted}>Reset</button>
        </div>
    );
};

export default MadLibArea;