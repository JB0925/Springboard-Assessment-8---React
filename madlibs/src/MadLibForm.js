import React, { useState } from "react";
import { useFormUpdate } from "./Hooks/hooks";
import "./MadLibForm.css";

/** The form used to send the data to the MadLibArea
 *  component.
 */
const MadLibForm = ({ addMadLib, updateSubmitted }) => {
    const [formData, updateForm, handleSubmit] = useFormUpdate();

    return (
        <form className="MadLibForm" onSubmit={(evt) => handleSubmit(addMadLib, updateSubmitted, evt)}>
            <label htmlFor="noun">Enter a Noun</label>
            <input
                id="noun"
                name="noun"
                type="text"
                value={formData.noun}
                onChange={updateForm}
                placeholder="i.e. bike, football, etc."
                required
            />
            <label htmlFor="nounTwo">Enter Another Noun</label>
            <input
                id="nounTwo"
                name="nounTwo"
                type="text"
                value={formData.nounTwo}
                onChange={updateForm}
                placeholder="i.e. bike, football, etc."
                required
            />
            <label htmlFor="adjective">Enter an Adjective</label>
            <input
                id="adjective"
                name="adjective"
                type="text"
                value={formData.adjective}
                onChange={updateForm}
                placeholder="i.e. funny, happy, ugly, etc."
                required
            />
            <label htmlFor="color">Enter a Color</label>
            <input
                id="color"
                name="color"
                type="text"
                value={formData.color}
                onChange={updateForm}
                placeholder="i.e. blue, yellow, etc."
                required
            />
            <button type="submit">Submit</button>
        </form>
    );
};

export default MadLibForm;