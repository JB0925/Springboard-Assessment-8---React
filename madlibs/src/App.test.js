import { render, screen, fireEvent } from '@testing-library/react';
import App from './App';
import MadLib from "./MadLib";
import MadLibForm from "./MadLibForm";
import MadLibArea from "./MadLibArea";

describe("Smoke tests for the Madlibs app - do the components render as they should?", () => {
  test('renders the App component', () => {
    render(<App />);
  });

  test('renders the MadLib component', () => {
    render(<MadLib />);
  });

  test('renders the MadLibForm component', () => {
    render(<MadLibForm />);
  });

  test('renders the MadLibArea component', () => {
    render(<MadLibArea />);
  });
});

describe('Snapshot tests - does each component match the snapshot?', () => {
  test('The App component matches the snapshot', () => {
    const { asFragment } = render(<App />);
    expect(asFragment()).toMatchSnapshot();
  });

  test('The MadLib component matches the snapshot', () => {
    const { asFragment } = render(<MadLib />);
    expect(asFragment()).toMatchSnapshot();
  });

  test('The MadLibForm component matches the snapshot', () => {
    const { asFragment } = render(<MadLibForm />);
    expect(asFragment()).toMatchSnapshot();
  });

  test('The MadLibArea component matches the snapshot', () => {
    const { asFragment } = render(<MadLibArea />);
    expect(asFragment()).toMatchSnapshot();
  });
});


