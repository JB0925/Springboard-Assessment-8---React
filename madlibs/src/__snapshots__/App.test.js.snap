// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot tests - does each component match the snapshot? The App component matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <h1>
      Madlibs!
    </h1>
    <div
      class="MadLibArea"
    >
      <form
        class="MadLibForm"
      >
        <label
          for="noun"
        >
          Enter a Noun
        </label>
        <input
          id="noun"
          name="noun"
          placeholder="i.e. bike, football, etc."
          required=""
          type="text"
          value=""
        />
        <label
          for="nounTwo"
        >
          Enter Another Noun
        </label>
        <input
          id="nounTwo"
          name="nounTwo"
          placeholder="i.e. bike, football, etc."
          required=""
          type="text"
          value=""
        />
        <label
          for="adjective"
        >
          Enter an Adjective
        </label>
        <input
          id="adjective"
          name="adjective"
          placeholder="i.e. funny, happy, ugly, etc."
          required=""
          type="text"
          value=""
        />
        <label
          for="color"
        >
          Enter a Color
        </label>
        <input
          id="color"
          name="color"
          placeholder="i.e. blue, yellow, etc."
          required=""
          type="text"
          value=""
        />
        <button
          type="submit"
        >
          Submit
        </button>
      </form>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Snapshot tests - does each component match the snapshot? The MadLib component matches the snapshot 1`] = `
<DocumentFragment>
  <p
    class="MadLib"
  />
</DocumentFragment>
`;

exports[`Snapshot tests - does each component match the snapshot? The MadLibArea component matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="MadLibArea"
  >
    <form
      class="MadLibForm"
    >
      <label
        for="noun"
      >
        Enter a Noun
      </label>
      <input
        id="noun"
        name="noun"
        placeholder="i.e. bike, football, etc."
        required=""
        type="text"
        value=""
      />
      <label
        for="nounTwo"
      >
        Enter Another Noun
      </label>
      <input
        id="nounTwo"
        name="nounTwo"
        placeholder="i.e. bike, football, etc."
        required=""
        type="text"
        value=""
      />
      <label
        for="adjective"
      >
        Enter an Adjective
      </label>
      <input
        id="adjective"
        name="adjective"
        placeholder="i.e. funny, happy, ugly, etc."
        required=""
        type="text"
        value=""
      />
      <label
        for="color"
      >
        Enter a Color
      </label>
      <input
        id="color"
        name="color"
        placeholder="i.e. blue, yellow, etc."
        required=""
        type="text"
        value=""
      />
      <button
        type="submit"
      >
        Submit
      </button>
    </form>
  </div>
</DocumentFragment>
`;

exports[`Snapshot tests - does each component match the snapshot? The MadLibForm component matches the snapshot 1`] = `
<DocumentFragment>
  <form
    class="MadLibForm"
  >
    <label
      for="noun"
    >
      Enter a Noun
    </label>
    <input
      id="noun"
      name="noun"
      placeholder="i.e. bike, football, etc."
      required=""
      type="text"
      value=""
    />
    <label
      for="nounTwo"
    >
      Enter Another Noun
    </label>
    <input
      id="nounTwo"
      name="nounTwo"
      placeholder="i.e. bike, football, etc."
      required=""
      type="text"
      value=""
    />
    <label
      for="adjective"
    >
      Enter an Adjective
    </label>
    <input
      id="adjective"
      name="adjective"
      placeholder="i.e. funny, happy, ugly, etc."
      required=""
      type="text"
      value=""
    />
    <label
      for="color"
    >
      Enter a Color
    </label>
    <input
      id="color"
      name="color"
      placeholder="i.e. blue, yellow, etc."
      required=""
      type="text"
      value=""
    />
    <button
      type="submit"
    >
      Submit
    </button>
  </form>
</DocumentFragment>
`;
